Source/DataModels/Resources/ResourceNavMesh.h

```diff
@@ -0,0 +1,322 @@
+#pragma once
+#include "Resource.h"
+//#include "Recast/SampleInterfaces.h"
+//#include "Utils/Buffer.h"
+#include <vector>
+
+
+class ResourceNavMesh : virtual public Resource
+{
+public:
+	ResourceNavMesh();
+	virtual ~ResourceNavMesh() override;
+
+	//bool Build(Scene* scene);
+	//void DrawGizmos(Scene* scene);	
+	//void Load(Buffer<char>& buffer);
+	//void CleanUp();
+	//Buffer<char> Save();
+
+	//bool IsGenerated() const;	
+	//dtCrowd* GetCrowd() const;
+	//dtNavMeshQuery* GetNavMeshQuery() const;
+	//dtNavMesh* GetNavMesh() const;
+	//dtTileCache* GetTileCache() const;
+
+	float GetAgentHeight() const;
+	float GetAgentRadius() const;
+	float GetAgentMaxClimb() const;
+	float GetAgentMaxSlope() const;
+	float GetCellSize() const;
+	float GetCellHeight() const;
+	float GetEdgeMaxError() const;
+
+	void SetAgentHeight(float agentHeight);
+	void SetAgentRadius(float agentRadius);
+	void SetAgentMaxClimb(float agentMaxClimb);
+	void SetAgentMaxSlope(float agentMaxSlope);
+	void SetCellSize(float cellSize);
+	void SetCellHeight(float cellHeight);
+	void SetEdgeMaxError(float edgeMaxError);
+
+	int GetRegionMinSize() const;
+	int GetRegionMergeSize() const;
+	int GetPartitionType() const;
+	int GetEdgeMaxLen() const;
+	int GetVertsPerPoly() const;
+	int GetDetailSampleDist() const;
+	int GetDetailSampleMaxError() const;
+	int GetTileSize() const;
+	int GetMaxTiles() const;
+	int GetMaxPolysPerTile() const;
+
+	void SetRegionMinSize(int regionMinSize);
+	void SetRegionMergeSize(int regionMergeSize);
+	void SetPartitionType(int partitionType);
+	void SetEdgeMaxLen(int edgeMaxLen);
+	void SetVertsPerPoly(int vertsPerPoly);
+	void SetDetailSampleDist(int detailSampleDist);
+	void SetDetailSampleMaxError(int detailSampleMaxError);
+	void SetTileSize(int tileSize);
+	void SetMaxTiles(int maxTiles);
+	void SetMaxPolysPerTile(int maxPolysPerTile);
+
+	bool GetKeepInterResults() const;
+
+	void SetKeepInterResults(bool keepInterResults);
+
+	unsigned char GetNavMeshDrawFlags() const;
+	
+	void SetNavMeshDrawFlags(unsigned char navMeshDrawFlags);
+
+public:
+	enum DrawMode
```

Comment:
Prefer use of scoped enums
```suggestion
	enum class DrawMode
```